---
editor_options: 
  markdown: 
    wrap: 72
---

|             |
|-------------|
| tle: "ldsR" |

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
library(ldsR)
library(testthat)
library(fs)
```

# Introduction

ldsR was designed to provide access to the main functions of the ldsc
python package inside R. An often cumbersome part of working with
summary statistics is the need to switch between R, Python and the
command line. This package moves some of that work to inside R.

# Use ldsR to directly estimate heritability inside R

Tired of preparing your files for ldsc? Use ldsR to directly estimate
heritability inside R.

`ldsc_h2()` has only one required argument, a dataframe with the columns
`SNP`, `Z` and `N`. The `ref-ld-chr` and `w-ld-chr` files are shipped
with the ldsR package, but you can also provide your own. By default
the, ldsR uses the ldscores derived from the European subset of 1000
Genomes.

```{r}
# example file, 100 000 rows to be lightweight.
df <- arrow::read_parquet(system.file("extdata", "sumstats.parquet", package = "ldsR"))


tdf <- dplyr::rename(df, Z = Z.x, N = N.x)
ldsc_h2(tdf)
```

# Genetic correlations

To calculate the genetic correlation between two traits, pass both
data.frames to `ldsc_rg()`. Note that there are now five mandatory
columns in each data.frame:

`SNP` `Z` `N` `A1` `A2`.

`A1` and `A2` are required to align the direction of the `Z` value
across two summary statistics.

```{r}
sumstat1 <- dplyr::rename(df, Z = Z.x, N = N.x) |> 
  dplyr::mutate(A1 = "A", A2 = "G")
sumstat2 <- dplyr::rename(df, Z = Z.y, N = N.y) |> 
  dplyr::mutate(A1 = "A", A2 = "G")

ldsc_rg(sumstat1, sumstat2)

```

`ldsc_rg()` provides an interface to easily run make genetic
correlations with an index trait.

for `sumstats2` you can provide a list of data.frames, and ldsc_rg will
calculate the genetic correlation with each summary statistic in the
list. If you provide a named list, the names will be saved in the
`trait2` column.

```{r}
sumstats <- list(
  "trait1" = sumstat2,
  "trait2" = sumstat2,
  "trait4" = sumstat2
)
          
ldsc_rg(sumstat1, sumstats)
        
```

# Partitioned heritability

To estimate the partitioned heritability, `partitioned_h2()` is
available, which requires the `ldscore_dir` argument, a directory
containing the partitioned ldscore files `ld.parquet` and
`annot.parquet` which has the M50 and M values. OBS, for overlapping
annotations, the functionality to adjust the enrichment estimate has not
yet been implemented. Therefore the enrichment estimate should not be
used.



```{r, eval = FALSE}
partitioned_h2(
  sumstat1,
  ldscore_dir = test_path("testdata/baseline")
)
```

# Cell-type analysis
Another common usage is a version of partitioned heritability sometimes referred to 
as "cell-type analysis".
Here we would like to estimate the significance of a heritability enrichment in a specific
set of genomic locations (for example specifically expressed genes in neurons), while
adjusting for another set of of annotations (all genes, genes expressed in brain, 
transcription factor binding sites etc). Typically, we will have several different
cell-types to test. We pass the annotations that should be in each model to
`covariate_dir`, and the ldscores for the cell-type to ldscore_dir


```{r, eval=FALSE}
celltype_analysis(
  sumstat1,
  covariate_dir = test_path("testdata/baseline"),
  ldscore_dir = test_path("testdata/superclusters")
)

```

