<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title> • ldsR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">ldsR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/ldsR.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/Ararder/ldsR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1></h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/Ararder/ldsR/blob/main/vignettes/articles/ldsR.Rmd" class="external-link"><code>vignettes/articles/ldsR.Rmd</code></a></small>
      <div class="d-none name"><code>ldsR.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/Ararder/ldsR" class="external-link">ldsR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://testthat.r-lib.org" class="external-link">testthat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://fs.r-lib.org" class="external-link">fs</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>ldsR exposes five functions:</p>
<p>1. <code><a href="../reference/celltype_analysis.html">celltype_analysis()</a></code> to estimate partitioned
heritability for many annotations one at a time, while adjusting for a
common set of annotations</p>
<p>2. <code><a href="../reference/ldsc_h2.html">ldsc_h2()</a></code> to estimate heritability for a single
trait</p>
<p>3. <code><a href="../reference/ldsc_rg.html">ldsc_rg()</a></code> to estimate the genetic correlation between
two traits</p>
<p>4. <code><a href="../reference/munge.html">munge()</a></code> to apply quality control filters to a summary
statistics file</p>
<p>6. <code><a href="../reference/partitioned_h2.html">partitioned_h2()</a></code> to estimate partitioned heritability
- OBS, correct enrichemnt estimates for overlapping annotations has not
yet been implemented.</p>
</div>
<div class="section level2">
<h2 id="use-ldsr-to-directly-estimate-heritability-inside-r">Use ldsR to directly estimate heritability inside R<a class="anchor" aria-label="anchor" href="#use-ldsr-to-directly-estimate-heritability-inside-r"></a>
</h2>
<p>Estimate heritability with one line inside R.</p>
<p><code><a href="../reference/ldsc_h2.html">ldsc_h2()</a></code> has only one required argument, a dataframe
with the columns <code>SNP</code>, <code>Z</code> and <code>N</code>.
The <code>ref-ld-chr</code> and <code>w-ld-chr</code> files are shipped
with the ldsR package, but you can also provide your own. By default
ldsR uses the ldscores derived from the European subset of 1000
Genomes.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># example file, 100 000 rows to be lightweight.</span></span>
<span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu">arrow</span><span class="fu">::</span><span class="fu"><a href="https://arrow.apache.org/docs/r/reference/read_parquet.html" class="external-link">read_parquet</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"sumstats.parquet"</span>, package <span class="op">=</span> <span class="st">"ldsR"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">tdf</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span><span class="va">df</span>, Z <span class="op">=</span> <span class="va">Z.x</span>, N <span class="op">=</span> <span class="va">N.x</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/ldsc_h2.html">ldsc_h2</a></span><span class="op">(</span><span class="va">tdf</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 × 6</span></span></span>
<span><span class="co">#&gt;      h2  h2_se   int int_se mean_chi2 lambda_gc</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> 0.371 0.035<span style="text-decoration: underline;">8</span>  1.09 0.032<span style="text-decoration: underline;">8</span>      2.14      1.80</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="genetic-correlations">Genetic correlations<a class="anchor" aria-label="anchor" href="#genetic-correlations"></a>
</h2>
<p>To calculate the genetic correlation between two traits, pass both
data.frames to <code><a href="../reference/ldsc_rg.html">ldsc_rg()</a></code>. Note that there are now five
mandatory columns in each data.frame:</p>
<p><code>SNP</code> <code>Z</code> <code>N</code> <code>A1</code>
<code>A2</code>.</p>
<p><code>A1</code> and <code>A2</code> are required to align the
direction of the <code>Z</code> value across two summary statistics.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sumstat1</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span><span class="va">df</span>, Z <span class="op">=</span> <span class="va">Z.x</span>, N <span class="op">=</span> <span class="va">N.x</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>A1 <span class="op">=</span> <span class="st">"A"</span>, A2 <span class="op">=</span> <span class="st">"G"</span><span class="op">)</span></span>
<span><span class="va">sumstat2</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span><span class="va">df</span>, Z <span class="op">=</span> <span class="va">Z.y</span>, N <span class="op">=</span> <span class="va">N.y</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>A1 <span class="op">=</span> <span class="st">"A"</span>, A2 <span class="op">=</span> <span class="st">"G"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/ldsc_rg.html">ldsc_rg</a></span><span class="op">(</span><span class="va">sumstat1</span>, <span class="va">sumstat2</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> 0 SNPs were removed when merging summary statistics</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 × 14</span></span></span>
<span><span class="co">#&gt;   trait2    rg  rg_se h2_trait1 h2_trait_se h2_trait2 h2_trait2_se  gcov gcov_se</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span>      1 0.734 0.049<span style="text-decoration: underline;">6</span>     0.371      0.035<span style="text-decoration: underline;">8</span>    0.073<span style="text-decoration: underline;">2</span>      0.006<span style="text-decoration: underline;">85</span> 0.121  0.011<span style="text-decoration: underline;">3</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 5 more variables: gcov_int &lt;dbl&gt;, gcov_int_se &lt;dbl&gt;, mean_z1z2 &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   z &lt;dbl&gt;, p &lt;dbl&gt;</span></span></span></code></pre></div>
<p><code><a href="../reference/ldsc_rg.html">ldsc_rg()</a></code> provides an interface to easily run make
genetic correlations with an index trait.</p>
<p>for <code>sumstats2</code> you can provide a list of data.frames, and
ldsc_rg will calculate the genetic correlation with each summary
statistic in the list. If you provide a named list, the names will be
saved in the <code>trait2</code> column.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sumstats</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  <span class="st">"trait1"</span> <span class="op">=</span> <span class="va">sumstat2</span>,</span>
<span>  <span class="st">"trait2"</span> <span class="op">=</span> <span class="va">sumstat2</span>,</span>
<span>  <span class="st">"trait4"</span> <span class="op">=</span> <span class="va">sumstat2</span></span>
<span><span class="op">)</span></span>
<span>          </span>
<span><span class="fu"><a href="../reference/ldsc_rg.html">ldsc_rg</a></span><span class="op">(</span><span class="va">sumstat1</span>, <span class="va">sumstats</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> 0 SNPs were removed when merging summary statistics</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> 0 SNPs were removed when merging summary statistics</span></span>
<span><span class="co">#&gt; ⠙ 2/3 ETA:  2s | Computing genetic correlations...</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> 0 SNPs were removed when merging summary statistics</span></span>
<span><span class="co">#&gt; ⠙ 2/3 ETA: 2s | Computing genetic correlations...</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 3 × 14</span></span></span>
<span><span class="co">#&gt;   trait2    rg  rg_se h2_trait1 h2_trait_se h2_trait2 h2_trait2_se  gcov gcov_se</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> trait1 0.734 0.049<span style="text-decoration: underline;">6</span>     0.371      0.035<span style="text-decoration: underline;">8</span>    0.073<span style="text-decoration: underline;">2</span>      0.006<span style="text-decoration: underline;">85</span> 0.121  0.011<span style="text-decoration: underline;">3</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> trait2 0.734 0.049<span style="text-decoration: underline;">6</span>     0.371      0.035<span style="text-decoration: underline;">8</span>    0.073<span style="text-decoration: underline;">2</span>      0.006<span style="text-decoration: underline;">85</span> 0.121  0.011<span style="text-decoration: underline;">3</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> trait4 0.734 0.049<span style="text-decoration: underline;">6</span>     0.371      0.035<span style="text-decoration: underline;">8</span>    0.073<span style="text-decoration: underline;">2</span>      0.006<span style="text-decoration: underline;">85</span> 0.121  0.011<span style="text-decoration: underline;">3</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 5 more variables: gcov_int &lt;dbl&gt;, gcov_int_se &lt;dbl&gt;, mean_z1z2 &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   z &lt;dbl&gt;, p &lt;dbl&gt;</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="munge">Munge<a class="anchor" aria-label="anchor" href="#munge"></a>
</h2>
<p>The <code><a href="../reference/munge.html">munge()</a></code> function can be used to mimic some of the
filters of the munge_sumstats.py file from ldsc.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/munge.html">munge</a></span><span class="op">(</span><span class="va">sumstat1</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Removed 0 rows after filtering on reference panel</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Removed 0 rows with duplicated RSIDs</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Removed 0 rows due to strand ambigious alleles</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Removed 0 rows with a sample size smaller than 87096</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 100,000 × 5</span></span></span>
<span><span class="co">#&gt;    SNP             Z      N A1    A2   </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> rs3094315   0.289 <span style="text-decoration: underline;">126</span>849 A     G    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> rs3131972   0.269 <span style="text-decoration: underline;">126</span>849 A     G    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> rs3131969  -<span style="color: #BB0000;">0.074</span> <span style="text-decoration: underline;">126</span>849 A     G    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> rs1048488   0.273 <span style="text-decoration: underline;">126</span>849 A     G    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> rs3115850   0.263 <span style="text-decoration: underline;">126</span>849 A     G    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> rs2286139   0.034 <span style="text-decoration: underline;">126</span>849 A     G    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> rs12562034  0.346 <span style="text-decoration: underline;">118</span>583 A     G    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> rs4040617   0.13  <span style="text-decoration: underline;">127</span>964 A     G    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> rs2980300  -<span style="color: #BB0000;">0.303</span> <span style="text-decoration: underline;">127</span>964 A     G    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> rs4970383  -<span style="color: #BB0000;">2.37</span>  <span style="text-decoration: underline;">114</span>957 A     G    </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 99,990 more rows</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="the-updated-ldscore-format">The updated ldscore format<a class="anchor" aria-label="anchor" href="#the-updated-ldscore-format"></a>
</h2>
<p>ldsR uses an updated format for ldscores, to reduce size and file
complexity. To run [partitioned_h2()] or [celltype_analysis()], you
provide a filepath to a directory containing two files:
<code>ld.parquet</code> and <code>annot.parquet</code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">fs</span><span class="fu">::</span><span class="fu"><a href="https://fs.r-lib.org/reference/dir_tree.html" class="external-link">dir_tree</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"baseline1.1_test"</span>, package <span class="op">=</span> <span class="st">"ldsR"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #0000BB; font-weight: bold;">/home/runner/work/_temp/Library/ldsR/extdata/baseline1.1_test</span></span></span>
<span><span class="co">#&gt; ├── annot.parquet</span></span>
<span><span class="co">#&gt; └── ld.parquet</span></span></code></pre></div>
<div class="section level3">
<h3 id="ld-parquet">ld.parquet<a class="anchor" aria-label="anchor" href="#ld-parquet"></a>
</h3>
<p><code>ld.parquet</code> contains the LD scores for each
annotation</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">arrow</span><span class="fu">::</span><span class="fu"><a href="https://arrow.apache.org/docs/r/reference/read_parquet.html" class="external-link">read_parquet</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"baseline1.1_test/ld.parquet"</span>, package <span class="op">=</span> <span class="st">"ldsR"</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice_head</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 5 × 5</span></span></span>
<span><span class="co">#&gt;   SNP     baseL2 Coding_UCSC.bedL2 Conserved_LindbladTo…¹ H3K4me1_peaks_Trynka…²</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> rs3094…   80.8             0.175                  0.392                   30.8</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> rs3131…   80.9             0.18                   0.394                   30.8</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> rs3131…   90.3             0.312                  0.519                   34.6</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> rs1048…   80.7             0.177                  0.394                   30.7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> rs3115…   80.5             0.144                  0.374                   30.6</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ abbreviated names: ¹​Conserved_LindbladToh.bedL2,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ²​H3K4me1_peaks_Trynka.bedL2</span></span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="annot-parquet">annot.parquet<a class="anchor" aria-label="anchor" href="#annot-parquet"></a>
</h3>
<p><code>annot.parquet</code> contains the annotations and M50 and M
values.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">arrow</span><span class="fu">::</span><span class="fu"><a href="https://arrow.apache.org/docs/r/reference/read_parquet.html" class="external-link">read_parquet</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"baseline1.1_test/annot.parquet"</span>, package <span class="op">=</span> <span class="st">"ldsR"</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice_head</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">5</span><span class="op">)</span> </span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 × 3</span></span></span>
<span><span class="co">#&gt;   annot                           m50       m</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> baseL2                      5<span style="text-decoration: underline;">961</span>159 9<span style="text-decoration: underline;">997</span>231</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Coding_UCSC.bedL2             <span style="text-decoration: underline;">85</span>095  <span style="text-decoration: underline;">160</span>751</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Conserved_LindbladToh.bedL2  <span style="text-decoration: underline;">153</span>113  <span style="text-decoration: underline;">285</span>337</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> H3K4me1_peaks_Trynka.bedL2  1<span style="text-decoration: underline;">012</span>339 1<span style="text-decoration: underline;">734</span>918</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="partitioned-heritability">Partitioned heritability<a class="anchor" aria-label="anchor" href="#partitioned-heritability"></a>
</h2>
<p>To estimate the partitioned heritability,
<code><a href="../reference/partitioned_h2.html">partitioned_h2()</a></code> is available, which requires the
<code>ldscore_dir</code> argument, a directory containing the
partitioned ldscore files <code>ld.parquet</code> and
<code>annot.parquet</code> which has the M50 and M values. OBS, for
overlapping annotations, the functionality to adjust the enrichment
estimate has not yet been implemented. Therefore the enrichment estimate
should not be used.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/partitioned_h2.html">partitioned_h2</a></span><span class="op">(</span></span>
<span>  <span class="va">sumstat1</span>,</span>
<span>  ldscore_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"baseline1.1_test"</span>, package <span class="op">=</span> <span class="st">"ldsR"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Removed 449 rows after merging with weights</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Removed 5961 rows after merging with ldscores</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 × 8</span></span></span>
<span><span class="co">#&gt;   annot                            coef coef_se enrich   prop     z   tot tot_se</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Conserved_LindbladToh.bedL2   8.54<span style="color: #949494;">e</span><span style="color: #BB0000;">-7</span> 2.55<span style="color: #949494;">e</span><span style="color: #BB0000;">-7</span> 18.2   0.021<span style="text-decoration: underline;">2</span>  3.35 0.339 0.038<span style="text-decoration: underline;">0</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> baseL2                        2.65<span style="color: #949494;">e</span><span style="color: #BB0000;">-8</span> 8.51<span style="color: #949494;">e</span><span style="color: #BB0000;">-9</span>  0.563 0.827   3.12 0.339 0.038<span style="text-decoration: underline;">0</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> H3K4me1_peaks_Trynka.bedL2    6.92<span style="color: #949494;">e</span><span style="color: #BB0000;">-8</span> 4.66<span style="color: #949494;">e</span><span style="color: #BB0000;">-8</span>  1.47  0.140   1.48 0.339 0.038<span style="text-decoration: underline;">0</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> Coding_UCSC.bedL2            -<span style="color: #BB0000;">2.28</span><span style="color: #949494;">e</span><span style="color: #BB0000;">-7</span> 1.61<span style="color: #949494;">e</span><span style="color: #BB0000;">-7</span> -<span style="color: #BB0000;">4.85</span>  0.011<span style="text-decoration: underline;">8</span> -<span style="color: #BB0000;">1.42</span> 0.339 0.038<span style="text-decoration: underline;">0</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="cell-type-analysis">Cell-type analysis<a class="anchor" aria-label="anchor" href="#cell-type-analysis"></a>
</h2>
<p>Another common usage is a version of partitioned heritability
sometimes referred to as “cell-type analysis”.</p>
<p>For [celltype_analysis()] one extra argument is required:
<code>covariate_dir</code>. <code>covariate_dir</code> will be used to
read in a set of LDscores that will be used as covariates when
estimating heritability enrichment. The ldscores in
<code>ldscores_dir</code> will be used to estimate the enrichment for
each annotation, one at a time, while adjusting for <em>all</em>
annotations in <code>covariate_dir</code>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/celltype_analysis.html">celltype_analysis</a></span><span class="op">(</span></span>
<span>  <span class="va">sumstat1</span>,</span>
<span>  covariate_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"baseline1.1_test"</span>, package <span class="op">=</span> <span class="st">"ldsR"</span><span class="op">)</span>,</span>
<span>  ldscore_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"superclusters"</span>, package <span class="op">=</span> <span class="st">"ldsR"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Removed 0 rows after merging with ldscores</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 × 8</span></span></span>
<span><span class="co">#&gt;   annot                     coef      coef_se enrich   prop      z   tot tot_se</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> amygdala_excitatory    3.02<span style="color: #949494;">e</span><span style="color: #BB0000;">-8</span> 0.000<span style="text-decoration: underline;">000</span>017<span style="text-decoration: underline;">4</span>  0.724 0.112   1.74  0.339 0.036<span style="text-decoration: underline;">1</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> astrocyte             -<span style="color: #BB0000;">1.45</span><span style="color: #949494;">e</span><span style="color: #BB0000;">-8</span> 0.000<span style="text-decoration: underline;">000</span>016<span style="text-decoration: underline;">0</span> -<span style="color: #BB0000;">0.338</span> 0.090<span style="text-decoration: underline;">0</span> -<span style="color: #BB0000;">0.911</span> 0.341 0.038<span style="text-decoration: underline;">7</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> bergmann_glia          4.39<span style="color: #949494;">e</span><span style="color: #BB0000;">-9</span> 0.000<span style="text-decoration: underline;">000</span>015<span style="text-decoration: underline;">4</span>  0.101 0.088<span style="text-decoration: underline;">1</span>  0.286 0.346 0.038<span style="text-decoration: underline;">3</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> cerebellar_inhibitory -<span style="color: #BB0000;">1.78</span><span style="color: #949494;">e</span><span style="color: #BB0000;">-8</span> 0.000<span style="text-decoration: underline;">000</span>013<span style="text-decoration: underline;">9</span> -<span style="color: #BB0000;">0.410</span> 0.087<span style="text-decoration: underline;">4</span> -<span style="color: #BB0000;">1.29</span>  0.343 0.039<span style="text-decoration: underline;">1</span></span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Arvid Harder.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
